org.eclipse.dawnsci.analysis.api/src/org/eclipse/dawnsci/analysis/api/processing/IOperationBean.java

```diff
@@ -11,43 +11,147 @@
  */
 public interface IOperationBean {
 
+	/**
+	 * Path to the node containing the unique keys for live processing
+	 * 
+	 * @param dataKey
+	 */
 	public void setDataKey(String dataKey);
 
+	/**
+	 * Set the path to the file to be processed
+	 * 
+	 * @param filePath
+	 */
 	public void setFilePath(String filePath);
 	
+	/**
+	 * Get the path to the file to be processed
+	 * 
+	 * @return filePath
+	 */
 	public String getFilePath();
 
+	/**
+	 * Set the absolute path the processed data should be saved as
+	 * 
+	 * @param outputFilePath
+	 */
 	public void setOutputFilePath(String outputFilePath);
 	
+	/**
+	 * Get the absolute path the processed data should be saved as
+	 * 
+	 * @return outputPath
+	 */
 	public String getOutputFilePath();
 
+	/**
+	 * Set the path to the dataset, or NXData node, to be processed
+	 * 
+	 * @param datasetPath
+	 */
 	public void setDatasetPath(String datasetPath);
 
+	/**
+	 * Set a string representation of the slicing of the dataset
+	 * 
+	 * @param slicing
+	 */
 	public void setSlicing(String slicing);
 
+	/**
+	 * Set the path to the NeXus file containing the processing chain
+	 * 
+	 * @param processingPath
+	 */
 	public void setProcessingPath(String processingPath);
 
+	/**
+	 * Set axes names for each dimension of the dataset set
+	 * 
+	 * @param axesNames
+	 */
 	public void setAxesNames(List<String>[] axesNames);
 
+	/**
+	 * Set whether the processing chain file should be deleted when processing is complete
+	 * 
+	 * @param deleteProcessingFile
+	 */
 	public void setDeleteProcessingFile(boolean deleteProcessingFile);
 
+	/**
+	 * Set the Xmx value used for the processing runner
+	 * 
+	 * @param xmx
+	 */
 	public void setXmx(String xmx);
 
+	/**
+	 * Set which dimensions of the dataset are the detector dimensions
+	 * 
+	 * Not needed if the scan rank is set
+	 * 
+	 * @param dataDimensions
+	 */
 	public void setDataDimensions(int[] dataDimensions);
 	
+	/**
+	 * Set the rank of the scan
+	 * 
+	 * Not needed if the data dimensions are set
+	 * 
+	 * @param scanRank
+	 */
 	public void setScanRank(Integer scanRank);
-
+	
+	/**
+	 * Sets whether the processed file should be readable while being written to
+	 * 
+	 * @param readable
+	 */
 	public void setReadable(boolean readable);
 	
+	/**
+	 * Set the name of the processing
+	 * 
+	 * @param name
+	 */
 	public void setName(String name);
 	
+	/**
+	 * Set the run directory for the logs
+	 * @param runDirectory
+	 */
 	public void setRunDirectory(String runDirectory);
 
+	/**
+	 * Set the number of core to be used
+	 * 
+	 * @param numberOfCores
+	 */
 	public void setNumberOfCores(int numberOfCores);
 
+	/**
+	 * Set the timeOut for live processing in ms
+	 * 
+	 * @param timeOut
+	 */
+	default void setTimeOut(int timeOut) {};
+	
+	/**
+	 * Set whether a link to the input data should be included in the processed
+	 * 
+	 * @param linkEntry
+	 */
+	default void setLinkParentEntry(boolean linkEntry){};
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this empty statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)